apps/bsd/src/components/ElectionConfiguration.tsx

```diff
@@ -1,54 +1,291 @@
-import React from 'react'
-import { useDropzone } from 'react-dropzone'
+import React, { useEffect, useState, useCallback } from 'react'
+import path from 'path'
 
+import styled from 'styled-components'
 import Prose from './Prose'
 import Main, { MainChild } from './Main'
 import MainNav from './MainNav'
 import Screen from './Screen'
 import Text from './Text'
+import Loading from './Loading'
+import FileInputButton, { HiddenFileInput } from './FileInputButton'
+import ChoiceButton from './ChoiceButton'
 
 import USBControllerButton from './USBControllerButton'
+import { UsbDriveStatus, getDevicePath } from '../lib/usbstick'
+import {
+  parseBallotExportPackageInfoFromFilename,
+  BALLOT_PACKAGE_FOLDER,
+} from '../util/filenames'
+import Button from './Button'
+
+const Image = styled.img`
+  margin: 0 auto -1.75rem;
+  max-width: 300px;
+`
+
+export const ChoicesGrid = styled.div`
+  display: grid;
+  grid-auto-rows: minmax(auto, 1fr);
+  grid-gap: 1rem;
+  p {
+    margin: 2px 0;
+  }
+`
+
+export const Container = styled.div`
+  margin: 0 10vw;
+`
+
+export const Footer = styled.div`
+  width: 100%;
+  justify-content: flex-end;
+  margin: 10px 0;
+  display: flex;
+  > button {
+    margin-left: 5px;
+  }
+`
+
+export const LinkLabel = styled.label`
+  color: rgb(0, 0, 238);
+  text-decoration: underline;
+  cursor: pointer;
+`
+
+export const Title = styled.strong`
+  text-transform: capitalize;
+`
 
 export interface Props {
-  acceptFiles(files: readonly File[]): void
+  acceptManuallyChosenFile(file: File): void
+  acceptAutomaticallyChosenFile(file: KioskBrowser.FileSystemEntry): void
+  usbDriveStatus: UsbDriveStatus
 }
 
-const ElectionConfiguration: React.FC<Props> = ({ acceptFiles }) => {
-  const { getRootProps, getInputProps, isDragActive } = useDropzone({
-    onDrop(files: readonly File[]) {
-      acceptFiles(files)
-    },
-    accept: ['application/json', 'application/zip'],
-  })
+const ElectionConfiguration: React.FC<Props> = ({
+  acceptAutomaticallyChosenFile,
+  acceptManuallyChosenFile,
+  usbDriveStatus,
+}) => {
+  const [foundFilenames, setFoundFilenames] = useState<
+    KioskBrowser.FileSystemEntry[]
+  >([])
+  const [loadingFiles, setLoadingFiles] = useState(false)
+  const [selectedFile, setSelectedFile] = useState<
+    KioskBrowser.FileSystemEntry | undefined
+  >(undefined)
+
+  const fetchFilenames = useCallback(async () => {
+    setLoadingFiles(true)
+    const usbPath = await getDevicePath()
+    try {
+      const files = await window.kiosk!.getFileSystemEntries(
+        path.join(usbPath!, BALLOT_PACKAGE_FOLDER)
+      )
+      setFoundFilenames(
+        files.filter((f) => f.type === 1 && f.name.endsWith('.zip'))
+      )
+      setLoadingFiles(false)
+    } catch (err) {
+      if (err.message.includes('ENOENT')) {
```

Comment:
This feels gross, but the error is wrapped in another error by kiosk-browser so there is no code on the error I catch here which is normally what you would check for "ENOENT" on